OpenDreamShared/Dream/IconAppearance.cs

```diff
@@ -12,6 +14,14 @@ public sealed class IconAppearance : IEquatable<IconAppearance> {
         [ViewVariables] public AtomDirection Direction;
         [ViewVariables] public Vector2i PixelOffset;
         [ViewVariables] public Color Color = Color.White;
+        /// <remarks>
+        /// An appearance can gain a color matrix filter by two possible forces: <br/>
+        /// 1. the /atom.color var is modified. <br/>
+        /// 2. the /atom.filters var gets a new filter of type "color". <br/>
+        /// DM crashes in some circumstances of this but we, as an extension :^), should try not to. <br/>
+        /// So, this exists as a way for the appearance to remember whether it's coloured by .color, specifically.
+        /// </remarks>
+        [ViewVariables] public DreamFilterColor? SillyColorFilter;
```

Comment:
I figure the reason this doesn't just replace `Color` entirely is so that we can avoid the slow filter-rendering path in the renderer when possible. Maybe leave a note about it being an optimization?

Reply:
It's not just an optimization, it's also what BYOND does! Using convertible matrices and then using the getter to read them back returns an RGB string.

The comment is in the summary of `TryRepresentMatrixAsRGBAColor`, but I can reference that material in this comment if you want :)